jsign-core/src/main/java/net/jsign/jca/AzureKeyVaultSigningService.java

```diff
@@ -152,9 +174,14 @@ public byte[] sign(SigningServicePrivateKey privateKey, String algorithm, byte[]
             throw new InvalidAlgorithmParameterException("Unsupported signing algorithm: " + algorithm);
         }
 
-        MessageDigest digest = DigestAlgorithm.of(algorithm.substring(0, algorithm.toLowerCase().indexOf("with"))).getMessageDigest();
+        DigestAlgorithm digestAlgorithm = DigestAlgorithm.of(algorithm.substring(0, algorithm.toLowerCase().indexOf("with")));
+        MessageDigest digest = digestAlgorithm.getMessageDigest();
         data = digest.digest(data);
 
+        if (digestAlgorithm == DigestAlgorithm.SHA1) {
```

Comment:
Better check that `"RSNULL".equals(alg)`, since this probably works with any algorithm and not just SHA-1. Also `createSha1RawDigest` could be inlined in the `sign` method.

Reply:
> Better check that `"RSNULL".equals(alg)`, since this probably works with any algorithm and not just SHA-1. Also `createSha1RawDigest` could be inlined in the `sign` method.

Will do.